src/qudi/hardware/interfuse/ni_scanning_probe_interfuse.py

```diff
@@ -458,18 +466,9 @@ def get_scan_data(self):
         @return (bool, ScanData): Failure indicator (fail=True), ScanData instance used in the scan
         #  TODO change interface
         """
-        # todo: get_scan data ussage for polling hw &iterating __read_pos seems sketchy
-        # => this hw file should implement it's own polling loop and provide updated ._scan_data
-        # when get_scan_data is called
         try:
-            if not self.is_scan_running or not self._ni_finite_sampling_io().is_running:
-                return self._scan_data
-            else:
-                # _stop_scan is called asynchronously. Thus .is_scan_running might be True, even if last data frame
-                # was already fetched. __scan_stopped is set by the polling thread, guaranteed to signal in time.
-                if not self.__scan_stopped:
-                    self._fetch_data_line()
-                return self._scan_data
+            with self._thread_lock_data:
+                return self._scan_data.copy()
```

Comment:
Is it safe to call .copy()? Eg. what happens if no scan was configured before, _scan_data could be == None.

Reply:
It indeed is not. So far the exception got swalloed by `self.log.exception("")`. But a simple `assert` could solve that and inform the user, that he should not call this without configuring a scan. `self._scan_data` is only None upon startup of the module.